### PR TITLE
Expand bbox to cover Earth for basemap search (but keep search center)

### PR DESCRIPTION
**Pull request by RZR-UA**

### CODE REVIEW

**Comment by RZR-UA** [comment]
https://github.com/osmandapp/OsmAnd/issues/18919

File: OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java [hunk 1]
```diff
@@ -604,7 +603,13 @@ public boolean search(final SearchPhrase phrase, final SearchResultMatcher resul
 					SearchPhraseDataType.POI);
 			String searchWord = phrase.getUnknownWordToSearch();
 			final NameStringMatcher nm = phrase.getMainUnknownNameStringMatcher();
-			QuadRect bbox = phrase.getFileRequest() != null ? phrase.getRadiusBBoxToSearch(BBOX_RADIUS_POI_IN_CITY) : phrase.getRadiusBBoxToSearch(BBOX_RADIUS_INSIDE);
+			QuadRect bbox = phrase.getRadiusBBoxToSearch(BBOX_RADIUS_POI_IN_CITY);
+			int centerX = (int) bbox.centerX();
+			int centerY = (int) bbox.centerY();
+			if (phrase.getFileRequest() == null) {
```
[the remark this answers is not in the document]
Comment (RZR-UA) [reply OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java]: refactored

File: OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java [hunk 2]
```diff
@@ -601,10 +600,18 @@ public boolean search(final SearchPhrase phrase, final SearchResultMatcher resul
...
 			String searchWord = phrase.getUnknownWordToSearch();
 			final NameStringMatcher nm = phrase.getMainUnknownNameStringMatcher();
-			QuadRect bbox = phrase.getFileRequest() != null ? phrase.getRadiusBBoxToSearch(BBOX_RADIUS_POI_IN_CITY) : phrase.getRadiusBBoxToSearch(BBOX_RADIUS_INSIDE);
+
+			QuadRect bbox = phrase.getRadiusBBoxToSearch(BBOX_RADIUS_POI_IN_CITY);
+			boolean worldWideSearch = phrase.getFileRequest() == null;
+			int centerX = (int) bbox.centerX();
+			int centerY = (int) bbox.centerY();
+			if (worldWideSearch) {
+				bbox.expand(0, 0, Integer.MAX_VALUE, Integer.MAX_VALUE);
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java:612]: bbox = new QuadRect… not expand

File: OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java [hunk 3]
```diff
@@ -601,10 +600,18 @@ public boolean search(final SearchPhrase phrase, final SearchResultMatcher resul
 
 			final BinaryMapIndexReader[] currentFile = new BinaryMapIndexReader[1];
 			Iterator<BinaryMapIndexReader> offlineIterator = phrase.getRadiusOfflineIndexes(BBOX_RADIUS,
-					SearchPhraseDataType.POI);
+					SearchPhraseDataType.POI); // WORLD obf(s) always + other obf(s) by radius
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java:603]: comment not needed - code should be fixed

File: OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java (hunk 2 above)
Comment (RZR-UA) [reply OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java]: fixed by another way

File: OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java (hunk 3 above)
Comment (RZR-UA) [reply OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java]: comment was useful but removed ok

File: OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java [hunk 4]
```diff
@@ -601,10 +600,18 @@ public boolean search(final SearchPhrase phrase, final SearchResultMatcher resul
...
 			String searchWord = phrase.getUnknownWordToSearch();
 			final NameStringMatcher nm = phrase.getMainUnknownNameStringMatcher();
-			QuadRect bbox = phrase.getFileRequest() != null ? phrase.getRadiusBBoxToSearch(BBOX_RADIUS_POI_IN_CITY) : phrase.getRadiusBBoxToSearch(BBOX_RADIUS_INSIDE);
+
+			QuadRect bbox = phrase.getRadiusBBoxToSearch(BBOX_RADIUS_POI_IN_CITY);
+			boolean worldWideSearch = phrase.getFileRequest() == null;
```
[the remark this answers is not in the document]
Comment (RZR-UA) [reply OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java]: refactored with isBasemap()

**Pull request merged**